src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -1414,17 +1427,34 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
 #endif
 	}
 	case SN_GetElement: {
+		int esize;
+		MonoType *etype;
+		
 		if (!is_element_type_primitive (fsig->params [0]))
 			return NULL;
 		MonoClass *arg_class = mono_class_from_mono_type_internal (fsig->params [0]);
-		MonoType *etype = mono_class_get_context (arg_class)->class_inst->type_argv [0];
+		
 		int size = mono_class_value_size (arg_class, NULL);
-		int esize = mono_class_value_size (mono_class_from_mono_type_internal (etype), NULL);
+		
+		if (vector_type->type == MONO_TYPE_GENERICINST)
+		{
+			etype = mono_class_get_context (arg_class)->class_inst->type_argv [0];
+			esize = mono_class_value_size (mono_class_from_mono_type_internal (etype), NULL);
+		}
+		else
+		{
+			// This exists to handle the static extension methods for Vector2/3/4 and Quaterion
+			// which live on System.Numerics.Vector
+
+			etype = m_class_get_byval_arg (mono_defaults.single_class);
+			esize = 4;
+		}
+
 		int elems = size / esize;
 		MONO_EMIT_NEW_BIALU_IMM (cfg, OP_COMPARE_IMM, -1, args [1]->dreg, elems);
 		MONO_EMIT_NEW_COND_EXC (cfg, GE_UN, "ArgumentOutOfRangeException");
-		int extract_op = type_to_xextract_op (arg0_type);
```

Comment:
`arg0_type` for this was just wrong before. `arg0` is `Vector128<T>` and we're extracting a `T`